src/makeTxn.ts

```diff
@@ -1,1448 +1,798 @@
-  AppOptInTxn,
-  AppCloseOutTxn,
-  AppClearStateTxn,
-  AppNoOpTxn,
-} from './types/transactions/index.js';
-import { RenameProperties, RenameProperty, Expand } from './types/utils.js';
+  SuggestedParams,
+  PaymentTransactionParams,
+  KeyRegistrationTransactionParams,
+  AssetConfigurationTransactionParams,
+  AssetTransferTransactionParams,
+  AssetFreezeTransactionParams,
+  ApplicationCallTransactionParams,
+} from './types/transactions/base.js';
+import { Address } from './encoding/address.js';
+
+/** Contains parameters common to every transaction type */
+export interface CommonTransactionParams {
+  /** Algorand address of sender */
+  sender: string | Address;
+  /** Suggested parameters relevant to the network that will accept this transaction */
+  suggestedParams: SuggestedParams;
+  /** Optional, arbitrary data to be stored in the transaction's note field */
+  note?: Uint8Array;
+  /**
+   * Optional, 32-byte lease to associate with this transaction.
+   *
+   * The sender cannot send another transaction with the same lease until the last round of original
+   * transaction has passed.
+   */
+  lease?: Uint8Array;
+  /** The Algorand address that will be used to authorize all future transactions from the sender, if provided. */
+  rekeyTo?: string | Address;
+}
 
 /**
- * makePaymentTxnWithSuggestedParams takes payment arguments and returns a Transaction object
- * @param sender - string representation of Algorand address of sender
- * @param receiver - string representation of Algorand address of recipient
- * @param amount - integer amount to send, in microAlgos
- * @param closeRemainderTo - optionally close out remaining account balance to this account, represented as string rep of Algorand address
- * @param note - uint8array of arbitrary data for sender to store
- * @param suggestedParams - a dict holding common-to-all-txns args:
- * fee - integer fee per byte, in microAlgos. for a flat fee, set flatFee to true
- * flatFee - bool optionally set this to true to specify fee as microalgos-per-txn
- *       If true, txn fee may fall below the ALGORAND_MIN_TX_FEE
- * firstValid - integer first protocol round on which this txn is valid
- * lastValid - integer last protocol round on which this txn is valid
- * genesisHash - string specifies hash genesis block of network in use
- * genesisID - string specifies genesis ID of network in use
- * @param rekeyTo - rekeyTo address, optional
+ * Create a new payment transaction
+ *
+ * @param options - Payment transaction parameters
  */
-export function makePaymentTxnWithSuggestedParams(
-  sender: PaymentTxn['sender'],
-  receiver: PaymentTxn['receiver'],
-  amount: PaymentTxn['amount'],
-  closeRemainderTo: PaymentTxn['closeRemainderTo'],
-  note: PaymentTxn['note'],
-  suggestedParams: MustHaveSuggestedParams<PaymentTxn>['suggestedParams'],
-  rekeyTo?: PaymentTxn['rekeyTo']
-) {
-  const o: PaymentTxn = {
+export function makePaymentTxnWithSuggestedParamsFromObject({
```

Comment:
I guess name change is to ensure user has to update/doesn't get caught by surprise?

Reply:
There's actually no name change, we just removed `makePaymentTxnWithSuggestedParams`, and `makePaymentTxnWithSuggestedParamsFromObject` was already defined below it

The diff is messy because it used to be the case that the `*FromObject` functions called the other ones